docs/knowledge-base/routing-sytem.md

```diff
@@ -0,0 +1,146 @@
+# Routing system in Drupal 8
+
+When a user visits a particular URL on a Drupal site, the path they are visiting is a route. 
+Drupal's routing system helps figure out which [controller][1] is responsible for responding to the route (URI) being requested. 
+Drupal matches the incoming request's path to a controller by looking through the routes that have been registered in active modules.
+
+## Why a new routing system
+
+In prior versions of Drupal, there was only hook_menu. Hook menu actually managed a few different features.
+In addition to handling incoming requests, it provided menu links, access control, action links, and a number of other features that are all very tightly coupled together.
+By using the Symfony2 Routing component, we are able to split out the route handling aspect, and get a much improved and feature-rich solution.
+Such as being able to create routes on more than just the path, for example make a request for JSON, XML or HTML while still using the same path. 
+
+
+## How the routing system works
+
+Drupal's routing system works with the Symfony HTTP Kernel. 
+The routing system is responsible for matching paths to controllers, and you define those relations in routes. You can pass on additional information to your controllers in the route. Access checking is integrated as well.
+[See drupal.org][5]
+
+## Structure of routes and parameters
+
+The route structure inside the *module.routing.yml* file is described in [drupal.org][3]
+
+## Adding new routes (static & dynamic) and altering existing ones
+
+
+### Adding a new static route
+
+1. In your example.routing.yml (let's use **example** module name as in drupal.org)
+    
+    Add your path, ( /something )
+        and in *defaults:* 
+            add the controller class and the method who will do something.
+            set a title
+        as *requirements:*
+            add permission   
+        
+
+```
+example.content:
+  path: '/example' 
+  defaults: 
+    _controller: '\Drupal\example\Controller\ExampleController::content' 
+    _title: 'Hello World'
+  requirements: 
+    _permission: 'access content' 
+```
+    
+
+2. In your src/Controller add your page controller class and your methods
+
+
+As basic route controller you can use Drupal\Core\Controller\ControllerBase.
+
+```namespace Drupal\example\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * An example controller.
+ */
+class ExampleController extends ControllerBase {
+
+  /**
+   * Returns a render-able array for a test page.
+   */
+  public function content() {
+    $build = [
+      '#markup' => $this->t('Hello World!'),
+    ];
+    return $build;
+  }
+}
+```
+
+### Adding a new dynamic route
+
+You can use the alterRoutes() method to add dynamic routes as well. 
```

Comment:
The given real example is a class... It should be something more practical, a use case.

Reply:
Agreed, then we should  change everything and write a tutorial .